src/Analyzers/VisualBasic/Analyzers/RemoveUnnecessaryByVal/VisualBasicRemoveUnnecessaryByValDiagnosticAnalyzer.vb

```diff
@@ -0,0 +1,46 @@
+﻿' Licensed to the .NET Foundation under one or more agreements.
+' The .NET Foundation licenses this file to you under the MIT license.
+' See the LICENSE file in the project root for more information.
+
+Imports System.Collections.Immutable
+Imports Microsoft.CodeAnalysis.CodeStyle
+Imports Microsoft.CodeAnalysis.Diagnostics
+Imports Microsoft.CodeAnalysis.VisualBasic.Syntax
+
+Namespace Microsoft.CodeAnalysis.VisualBasic.RemoveUnnecessaryByVal
+    <DiagnosticAnalyzer(LanguageNames.VisualBasic)>
+    Friend Class VisualBasicRemoveUnnecessaryByValDiagnosticAnalyzer
+        Inherits AbstractCodeStyleDiagnosticAnalyzer
+
+        Private Shared ReadOnly s_descriptor As New DiagnosticDescriptor(
+            id:=IDEDiagnosticIds.RemoveUnnecessaryByValDiagnosticId,
+            title:="",
+            messageFormat:="",
+            category:="",
+            defaultSeverity:=DiagnosticSeverity.Hidden,
+            isEnabledByDefault:=True,
+            description:="",
+            helpLinkUri:="",
+            WellKnownDiagnosticTags.Unnecessary)
+
```

Comment:
This should not be needed.
```suggestion
```

Reply:
If I remove this, what should I pass to the base constructor?
I cannot find out how I'd indicate:

- `defaultSeverity:=DiagnosticSeverity.Hidden`
- `WellKnownDiagnosticTags.Unnecessary`

with the other ctor overloads.

Reply:
DefaultSeverity for code style diagnostics is already hidden: http://sourceroslyn.io/#Microsoft.CodeAnalysis.Features/AbstractCodeStyleDiagnosticAnalyzer.cs,86

You don't need to pass the Unnecessary tag, instead directly access `UnnecessaryWithSuggestionDescriptor` in your analyzer: http://sourceroslyn.io/#Microsoft.CodeAnalysis.Features/AbstractCodeStyleDiagnosticAnalyzer.cs,22